test/units/module_utils/ec2/test_camel_to_snake.py

```diff
@@ -0,0 +1,35 @@
+# -*- coding: utf-8 -*-
+# (c) 2017, Will Thames <will.thames@xvt.com.au>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.compat.tests import unittest
+from ansible.module_utils.ec2 import _camel_to_snake
+
+EXPECTED_CAMELIZATION = {
+    'alllower': 'alllower',
+    'TwoWords': 'two_words',
+    'AllUpperAtEND': 'all_upper_at_end',
+    'AllUpperButPLURALs': 'all_upper_but_plurals',
```

Comment:
This clearly does fix uppercase plurals if there is something CamelCase preceding. AllUpperButPLURALs isn't treated the same as PLURALs though, since it isn't actually only uppercase except the last letter. For instance, ARNs becomes _arns. I don't know if there are any instances when this might occur. Maybe there's no reason to care about it.
Example:
```
(python3.5.2) shertel-OSX-2:ansible shertel$ git diff
diff --git a/test/units/module_utils/ec2/test_camel_to_snake.py b/test/units/module_utils/ec2/test_camel_to_snake.py
index cbdf8d5c9f..785686765f 100644
--- a/test/units/module_utils/ec2/test_camel_to_snake.py
+++ b/test/units/module_utils/ec2/test_camel_to_snake.py
@@ -25,6 +25,7 @@ EXPECTED_CAMELIZATION = {
     'AllUpperAtEND': 'all_upper_at_end',
     'AllUpperButPLURALs': 'all_upper_but_plurals',
     'TargetGroupARNs': 'target_group_arns',
+    'ARNs': 'arns'
 }
```
```
___________________________ CamelToSnakeTestCase.test_camel_to_snake ____________________________

self = <units.module_utils.ec2.test_camel_to_snake.CamelToSnakeTestCase testMethod=test_camel_to_snake>

    def test_camel_to_snake(self):
        for (k, v) in EXPECTED_CAMELIZATION.items():
>           self.assertEqual(_camel_to_snake(k), v)
E           AssertionError: '_arns' != 'arns'
```